src/OrleansCodeGenerator/GrainReferenceGenerator.cs

```diff
@@ -203,6 +205,19 @@ private static MemberDeclarationSyntax[] GenerateInvokeMethods(Type grainType, A
                     }
 
                     body.Add(SF.ExpressionStatement(invocation));
+
+                    if (isOneWayTask)
+                    {
+                        if (!method.ReturnType.IsInstanceOfType(TaskDone.Done))
+                        {
+                            throw new CodeGenerationException(
+                                $"Method {grainType.GetParseableName()}.{method.Name} is marked with [{nameof(OneWayAttribute)}], " +
+                                $"but has a return type which is not assignable from {TaskDone.Done.GetType()}");
```

Comment:
why are you using TaskDone.Done instead of just `typeof(Task)`? Am I missing and edge case? Is it because we might want to support `AsyncValue`/ `TaskAwaiter`/`INotifyCompletion` in te future so you are protecting from those changes?